django/core/validators.py

```diff
@@ -90,6 +83,68 @@ class URLValidator(RegexValidator):
         r"(?<!-)"  # can't end with a dash
         r"\.?"  # may have a trailing dot
     )
+    ascii_only_hostname_re = r"[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?"
+    ascii_only_domain_re = r"(?:\.(?!-)[a-zA-Z0-9-]{1,63}(?<!-))*"
+    ascii_only_tld_re = (
+        r"\."  # dot
+        r"(?!-)"  # can't start with a dash
+        r"(?:[a-zA-Z0-9-]{2,63})"  # domain label
+        r"(?<!-)"  # can't end with a dash
+        r"\.?"  # may have a trailing dot
+    )
+
+    max_length = 255
+
+    def __init__(self, **kwargs):
+        self.accept_idna = kwargs.pop("accept_idna", True)
+
+        if self.accept_idna:
+            self.regex = _lazy_re_compile(
+                self.hostname_re + self.domain_re + self.tld_re, re.IGNORECASE
+            )
+        else:
+            self.regex = _lazy_re_compile(
+                self.ascii_only_hostname_re
+                + self.ascii_only_domain_re
+                + self.ascii_only_tld_re,
+                re.IGNORECASE,
+            )
+        super().__init__(**kwargs)
+
+    def __call__(self, value):
+        if not isinstance(value, str) or len(value) > self.max_length:
+            raise ValidationError(self.message, code=self.code, params={"value": value})
+
+        if self.accept_idna:
+            super().__call__(value)
+        else:
+            if not value.isascii():
+                raise ValidationError(
+                    self.message, code=self.code, params={"value": value}
+                )
+            super().__call__(value)
+
+
+validate_domain_name = DomainNameValidator()
+
+
+@deconstructible
+class URLValidator(RegexValidator):
+    ul = "\u00a1-\uffff"  # Unicode letters range (must not be a raw string).
+
+    # IP patterns
+    ipv4_re = (
+        r"(?:0|25[0-5]|2[0-4][0-9]|1[0-9]?[0-9]?|[1-9][0-9]?)"
+        r"(?:\.(?:0|25[0-5]|2[0-4][0-9]|1[0-9]?[0-9]?|[1-9][0-9]?)){3}"
+    )
+    ipv6_re = r"\[[0-9a-f:.]+\]"  # (simple regex, validated later)
+
+    # Host patterns
+    hostname_re = DomainNameValidator.hostname_re
+    # Max length for domain name labels is 63 characters per RFC 1034 sec. 3.1
```

Comment:
```suggestion
    hostname_re = DomainNameValidator.hostname_re
```
As per the above comment which hopefully makes sense